internal/cmd/configuration/command_set.go

```diff
@@ -0,0 +1,119 @@
+package configuration
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/examples"
+	"github.com/confluentinc/cli/internal/pkg/form"
+	"github.com/confluentinc/cli/internal/pkg/output"
+)
+
+func (c *command) newSetCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set <config-field-1=value-1> ... [config-field-n=value-n]",
+		Short: "Set a field's value in ~/.confluent/config.json.",
+		Args:  cobra.MinimumNArgs(1),
+		RunE:  c.set,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Disable plugins by setting "disable_plugins" to true`,
+				Code: `confluent configuration set disable_plugins=true`,
+			},
+		),
+	}
+
+	return cmd
+}
+
+func (c *command) set(cmd *cobra.Command, args []string) error {
+	jsonFieldToType, jsonFieldToName := getSettableConfigFields(c.config)
+	keys, values, err := buildUpdates(args, jsonFieldToType)
+	if err != nil {
+		return err
+	}
+
+	updates := make([]string, len(args))
+
+	for i := range keys {
+		oldValue := reflect.ValueOf(c.config).Elem().FieldByName(jsonFieldToName[keys[i]])
+		switch v := values[i].(type) {
+		case bool:
+			if keys[i] == "disable_feature_flags" {
+				if ok, err := confirmSet(keys[i], form.NewPrompt()); err != nil {
+					return err
+				} else if !ok {
+					continue
+				}
+			}
+			oldValue.SetBool(v)
+		case string:
+			oldValue.SetString(v)
+		}
+		updates = append(updates, fmt.Sprintf(errors.UpdateSuccessMsg, "value", "config field", keys[i], values[i]))
```

Comment:
This seems like it has the potential to be misleading. If the user tries to update the `current_context` to a bad value, this line would report success.

But then we would error out on the `Validate()` all below since that function checks that the `current_context` is in the list of contexts.

Also, I think perhaps we should not let users update the context here since we already have `confluent context use`. Perhaps we could include a message directing them to use that instead?

Reply:
`updates` only print out if the `Validate()` and `Save()` functions return normally.

Yeah, I think if they attempt to set `current_context`, I'll just direct them to `context use`.

Reply:
Ah, I misread. Nice!